### PR TITLE
Improving hints (Lombiq Technologies: OCORE-21)

### PR DESCRIPTION
**Pull request by Piedone**
Improving UI hints that I found in need of improvements.

### CODE REVIEW

**Comment by hishamco** [comment]
Dean its yours for now ;)

File: src/OrchardCore.Modules/OrchardCore.Flows/Migrations.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public int UpdateFrom1()
             _contentDefinitionManager.AlterPartDefinition("BagPart", builder => builder
                 .Attachable()
                 .Reusable()
-                .WithDescription("Provides a collection behavior for your content item."));
+                .WithDescription("Provides a collection behavior for your content item, you'll be able to put other content items into this one."));
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Flows/Migrations.cs:31]:
```suggestion
                .WithDescription("Provides a collection behavior for your content item where you can place other content items"));
```

File: src/OrchardCore.Modules/OrchardCore.Users/Views/RegistrationSettings.Edit.cshtml [hunk 2]
```diff
@@ -90,7 +90,7 @@
         <input type="checkbox" class="custom-control-input" asp-for="NoUsernameForExternalUsers" data-external-registration />
         <span asp-validation-for="NoUsernameForExternalUsers"></span>
         <label class="custom-control-label" asp-for="NoUsernameForExternalUsers">@T["Do not ask username"]</label>
-        <span class="hint">— @T["When a new user logs in with an external provider, he is not required to provide a local username.You can customize how it looks by providing an IExternalLoginEventHandler or write a script."]</span>
+        <span class="hint">— @T["When a new user logs in with an external provider, he is not required to provide a local username. You can customize how it looks by providing an IExternalLoginEventHandler or write a script."]</span>
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Users/Views/RegistrationSettings.Edit.cshtml:93]:
```suggestion
        <span class="hint">— @T["When a new user logs in with an external provider, they are not required to provide a local username. You can customize how it looks by providing an IExternalLoginEventHandler or writing a script."]</span>
```

File: src/OrchardCore.Modules/OrchardCore.Widgets/Migrations.cs [hunk 3]
```diff
@@ -17,7 +17,7 @@ public int Create()
         {
             _contentDefinitionManager.AlterPartDefinition("WidgetsListPart", builder => builder
                 .Attachable()
-                .WithDescription("Provides a way to add widgets per content items.")
+                .WithDescription("Provides a way to add widgets to Layout zones just for a given content item.")
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Widgets/Migrations.cs:20]:
```suggestion
                .WithDescription("Provides a way to add widgets to Layout zones for a given content item.")
```

File: src/OrchardCore.Modules/OrchardCore.Layers/Views/LayerSettings.Edit.cshtml [hunk 4]
```diff
@@ -4,5 +4,8 @@
     <label asp-for="Zones">@T["Available zones for Layers"]</label>
     <input asp-for="Zones" class="form-control" type="text" />
     <span asp-validation-for="Zones"></span>
-    <span class="hint">@T["A comma separated list of values, representing the available zones for widgets."]</span>
+    <span class="hint">
+        @T["A comma separated list of values, representing the available zones for <a href=\"{0}\">widgets</a>. Note that you'll need to render these zones in the layout template, in themselves they won't work.",
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Layers/Views/LayerSettings.Edit.cshtml:8]:
```suggestion
        @T["A comma separated list of values, representing the available zones for <a href=\"{0}\">widgets</a>. Note that you'll need to render these zones in the layout template, as they do not automatically render.",
```

File: src/OrchardCore.Modules/OrchardCore.Users/Views/RegistrationSettings.Edit.cshtml (hunk 2 above)
Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Users/Views/RegistrationSettings.Edit.cshtml:93]: Do you think this is `You can customize how it looks by` or `You can customize how it works` ? @Piedone

Comment (Piedone) [reply src/OrchardCore.Modules/OrchardCore.Users/Views/RegistrationSettings.Edit.cshtml]: Good catch and yes, it indeed should rather reference "works". I've changed it.

File: src/OrchardCore.Modules/OrchardCore.Flows/Migrations.cs [hunk 5]
```diff
@@ -28,7 +28,7 @@ public int UpdateFrom1()
             _contentDefinitionManager.AlterPartDefinition("BagPart", builder => builder
                 .Attachable()
                 .Reusable()
-                .WithDescription("Provides a collection behavior for your content item."));
+                .WithDescription("Provides a collection behavior for your content item where you can place other content items"));
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Flows/Migrations.cs:31]:
```suggestion
                .WithDescription("Provides a collection behavior for your content item where you can place other content items."));
```

Comment (Piedone) [reply src/OrchardCore.Modules/OrchardCore.Flows/Migrations.cs:31]: Good catch!

**Pull request merged**